vsphere/data_source_vsphere_datastore_cluster.go

```diff
@@ -0,0 +1,35 @@
+package vsphere
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceVSphereDatastoreCluster() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceVSphereDatastoreClusterRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The name or absolute path to the datastore cluster.",
+			},
+			"datacenter_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The managed object ID of the datacenter the cluster is located in. Not required if using an absolute path.",
+			},
+		},
+	}
+}
+
+func dataSourceVSphereDatastoreClusterRead(d *schema.ResourceData, meta interface{}) error {
+	pod, err := resourceVSphereDatastoreClusterGetPodFromPath(meta, d.Get("name").(string), d.Get("datacenter_id").(string))
```

Comment:
Actually, a short-form name works regardless of where the datastore cluster is in inventory. Since we use `Finder` to ultimately look for most objects when searching by name or path, the same rule apply. See https://godoc.org/github.com/vmware/govmomi/find and https://github.com/vmware/govmomi/blob/master/govc/README.md#usage for more details on how paths can be searched on in govmomi.

Reply:
Got it. Looks good then!